### PR TITLE
LCD_Actor 

### PR DESCRIPTION
**Pull request by seinizumita**
The actor for the LCD display

Yusra and I added a function to auto-scroll any text longer than 16 characters and have it centered when possible.
Since we don't have both sensor and lcd, we weren't able to test both at the same time.

### CODE REVIEW

File: bellboy/actors/lead.py [hunk 1]
```diff
@@ -25,14 +36,27 @@ def startBellboyLead(self):
...
 
-        # request to setup sensor
-        self.send(
-            self.ultrasonic_sensor,
-            SensorMsg(SensorReq.SETUP, trigPin=23, echoPin=24, maxDepth_cm=200),
+        # setup actors, handle their responses
+        sensor_setup_msg = SensorMsg(
+            SensorReq.SETUP, trigPin=23, echoPin=24, maxDepth_cm=200
         )
+        lcd_setup_msg = LcdMsg(LcdReq.SETUP, defaultText="Welcome to" "\n" "Bellboy")
```
Comment (yozohu) [inline bellboy/actors/lead.py]: you should be able to take out the \n here

File: bellboy/actors/lead.py [hunk 2]
```diff
@@ -25,14 +36,27 @@ def startBellboyLead(self):
...
+
+        self.send(self.ultrasonic_sensor, sensor_setup_msg)
+        self.send(self.lcd, lcd_setup_msg)
+
         self.status = Response.STARTED
 
+        sleep(2)
```
Comment (yozohu) [inline bellboy/actors/lead.py]: can take these lines out, was just for testing purposes

File: bellboy/actors/lcd.py [hunk 3]
```diff
@@ -1,8 +1,215 @@
...
+        self.lcd = None
+        self.default_text = None
+
+    # --------------------------#
+    # STATE MODIFYING METHODS   #
+    # --------------------------#
+
+    def setup(self, default_text):
+        """setup LCD."""
+        # Create PCF8574 GPIO adapter.
```
Comment (yozohu) [inline bellboy/actors/lcd.py]: comment could be placed better, i.e. move this one to line 38

File: bellboy/actors/lcd.py [hunk 4]
```diff
@@ -1,8 +1,215 @@
...
+
+        self.lcd = Adafruit_CharLCD(pin_rs=0, pin_e=2, pins_db=[4, 5, 6, 7], GPIO=mcp)
+        mcp.output(3, 1)  # turn on LCD backlight
+        self.lcd.begin(16, 2)  # set number of LCD lines and columns
+
+        self.displayText("System Starting...", 2)
+        self.status = LcdResp.SET
+
+    def displayText(self, text, duration):
+        """Displays text on screen for duration of time, then returns to defualt message"""
```
Comment (yozohu) [inline bellboy/actors/lcd.py]: defualt

File: bellboy/actors/lcd.py [hunk 5]
```diff
@@ -1,8 +1,215 @@
...
+
+    def clear(self):
+        self.lcd.clear()
+
+        self.log.debug("cleared LCD")
+        self.status = Response.READY
+
+    def printText(self, text, duration, scroll=False):
+        """Prints text to the LCD. If scroll = true, Scrolls the text for duration if the text is too long.
+        Returns whether True if text was scrolled"""
```
Comment (yozohu) [inline bellboy/actors/lcd.py]: "Returns true if text was scrolled"

File: bellboy/utils/messages.py [hunk 6]
```diff
@@ -89,10 +95,39 @@ class SensorEvent(Enum):
...
+
+"""Lcd messages"""
+
+
+class LcdReq(Enum):
+    SETUP, DISPLAY, CLEAR = range(3)
+
+
+class LcdResp(Enum):
+    SET, DISPLAYING, SPILLING, CENTERING = range(4)
```
Comment (yozohu) [inline bellboy/utils/messages.py]: those last 2 arent used

**Review by yozohu — COMMENTED** [review]
minor comments
other than those looks good

**Review by RyanFleck — APPROVED** [review]
LGTM!

**Pull request merged**